orientation-sensor/OrientationSensor.https.html

```diff
@@ -0,0 +1,127 @@
+ <!DOCTYPE html>
+<meta charset="utf-8">
+<title>OrientationSensor Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="help" href="https://w3c.github.io/orientation-sensor/">
+<link rel="help" href="https://w3c.github.io/sensors/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/generic-sensor/generic-sensor-tests.js"></script>
+<div id="log"></div>
+
+<script>
+
+//IEEE 754: single precision retricts to 7 decimal digits
+//          double precision retricts to 15 decimal digits
+const float_precision = 1e-7;
+const double_precision = 1e-15;
+
+function create_matrix(quat) {
+  let X = quat[0];
+  let Y = quat[1];
+  let Z = quat[2];
+  let W = quat[3];
+  let mat = new Array(
+    1-2*Y*Y-2*Z*Z, 2*X*Y-2*Z*W, 2*X*Z+2*Y*W, 0,
+    2*X*Y+2*Z*W, 1-2*X*X-2*Z*Z, 2*Y*Z-2*X*W, 0,
+    2*X*Z-2*Y*W, 2*Y*Z+2*W*X, 1-2*X*X-2*Y*Y, 0,
+    0, 0, 0, 1
+  );
+  return mat;
+}
+
+function assert_array_approx_equals(actual, expected, epsilon) {
+  assert_true(actual.length === expected.length);
+  for (let i = 0; i < expected.length; i++) {
+    assert_approx_equals(actual[i], expected[i], epsilon)
+  }
+}
+
+function checkQuaternion(t, sensorType) {
+  let sensor = new sensorType();
+  sensor.onreading = t.step_func_done(() => {
+    assert_equals(sensor.quaternion.length, 4);
+    assert_true(sensor.quaternion instanceof Array);
+    sensor.stop();
+  });
+  sensor.onerror = t.step_func_done(unreached);
+  sensor.start();
+};
+
+function checkPopulateMatrix(t, sensorType) {
+  let sensor = new sensorType();
+  sensor.onerror = t.step_func_done(unreached);
+
+  //Throws with insufficient buffer space.
+  assert_throws({ name: 'TypeError' }, () => sensor.populateMatrix(new Float32Array(15)));
+
+  //Throws if no orientation data available.
+  assert_throws({ name: 'NotReadableError' }, () => sensor.populateMatrix(new Float32Array(16)));
+
+  if (window.SharedArrayBuffer) {
+    // Throws if passed SharedArrayBuffer view.
+    assert_throws({ name: 'TypeError' }, () => sensor.populateMatrix(new Float32Array(new SharedArrayBuffer(16))));
+  }
+
+  sensor.onreading = t.step_func_done(() => {
+    let quat = sensor.quaternion;
+    let mat_expect = create_matrix(quat);
+
+    // Works for all supported types.
+    let mat_32 = new Float32Array(16);
+    sensor.populateMatrix(mat_32);
+    assert_array_approx_equals(mat_32, mat_expect, float_precision);
+
+    let mat_64 = new Float64Array(16);
+    sensor.populateMatrix(mat_64);
+    assert_array_approx_equals(mat_64, mat_expect, double_precision);
```

Comment:
We found the test result is sometime failed due to lose precision issue in Float64Array and thought it was same as Float32Array. I will fixed it and file bug if the failure occurs again.